.github/workflows/build-images-docs-builder.yaml

```diff
@@ -90,7 +91,7 @@ jobs:
   update-pr:
     name: Update Pull Request with new image reference
     needs: build-and-push
-    if: needs.build-and-push.outputs.digest
+    if: github.repository_owner == 'cilium' && needs.build-and-push.outputs.digest
```

Comment:
Are this one, and the one below, necessary? If the owner is not `cilium`, then `build-and-push` will not run and `needs.build-and-push.outputs.digest` shouldn't resolve as `true`?

Looks good otherwise for adding the restriction. The workflow won't work on forks anyway, unless the owner has bothered setting up the relevant environment and secrets.

Reply:
oh! that's a good point. skipping `build-and-push` should be enough. updating the PR now...

Reply:
could you approve https://github.com/cilium/cilium/actions/runs/6007268461/job/16293139576?pr=27757 just to make sure it still works 🙏